src/StdIOCallback.cpp

```diff
@@ -102,6 +102,8 @@ uint32 StdIOCallback::read(void*Buffer,size_t Size)
   assert(File!=nullptr);
 
   size_t result = fread(Buffer, 1, Size, File);
+  if (feof(File) || ferror(File))
```

Comment:
Does this really make sense? If you're, say, at position 500 in a file of size 1000 & request to read 2KB, the file descriptor will have its EOF flag set, and you simply don't return the last 500 bytes that could & were actually read to the calling function, thereby losing data. You cannot expect a function to only request exactly as much as the file size indicates, and punishing it for it is not nice.

Or am I missing something here?

Reply:
I get all that. For libEBML/libMatroska this sounds OK. I'm just worried that if any other program is using `StdIOCallback` this would be a grave behavior change for them. I actually don't know if there are any; MKVToolNix has its own I/O class derived from `IOCallback`.